jsonrpc/jsonrpc.go

```diff
@@ -148,3 +167,33 @@ func listen(addr string, jsonRPCConfig config.JSONRPCConfig) (net.Listener, erro
 	}
 	return ln, err
 }
+
+// reference: https://github.com/evmos/ethermint/blob/fd8c2d25cf80e7d2d2a142e7b374f979f8f51981/server/util.go#L74
+func ConnectCometWS(cometRPCAddr, cometWSEndpoint string, logger log.Logger) *rpcclient.WSClient {
+	cometWSClient, err := rpcclient.NewWS(cometRPCAddr, cometWSEndpoint,
+		//TODO: make the following values configurable
+		rpcclient.MaxReconnectAttempts(256),
```

Comment:
@jasonsong0 
Both ReadWait and WriteWait values are ok to set zero?

When establishing a WebSocket client connection, you mentioned that the `readwait` and `writewait` should be set longer than the filter's deadline.

Reply:
@jasonsong0 
Please add the comment this line why we should set readwait and writewait longer than filter's deadline. 
e.g. `// When establishing a WebSocket client connection, the readwait and writewait should be set longer than the filter's deadline to avoid abnormal closure`

Reply:
zero means block until operation succeeds. 
It means no read, write deadline for the websocket connection.
That's [the default value](https://github.com/initia-labs/cometbft/blob/6c77a401128cb7dd8368ba8fbe7f30caf4fffa96/rpc/jsonrpc/client/ws_client.go#L24) for ws_client 

normally, readWait don't needed to be set longer than filter deadline. but pingPeriod should be set less than readWait if readWait is not 0. 
I found a situlation where the connection was closed and trying to reconnect even though the readWait is larger than the pingPeriod. To avoid abnormal closures, I changed it to zero.

Reply:
@jasonsong0 
Thanks for the detailed explanation.

Could you please add comments to the related code for the following points?
- That the ping period should be set less than readWait if readWait is not zero
- What happened if we don't do above.

I think it's important to document the reasoning behind these decisions for future reference.